### PR TITLE
docs: update doc around fetching unscheduled pods using --node=

### PR DESCRIPTION
**Pull request by CatherineF-dev**
**What this PR does / why we need it**:

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)* n/a

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kube-state-metrics/issues/2353

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -296,6 +296,12 @@ spec:
         - --node=""
 ```
 
+| KSM --node parameter | FieldSelector |
```
Comment (mrueg) [inline README.md:299]: Can we have a human readable column as well that explains what this setting does?

Comment (CatherineF-dev) [reply README.md]: Updated

**Comment by mrueg** [comment]
Sorry, I forgot: The change needs to be made in README.md.tpl and then you'd need to run the make target to regenerate the README file with gomplate.

File: README.md [hunk 2]
```diff
@@ -296,6 +296,12 @@ spec:
         - --node=""
 ```
 
+| KSM --node parameter | FieldSelector | Note |
+| ---                  | --- | --- |
+| --node="node1"       | kubectl get pods --field-selector spec.nodeName="node1" | Fetch all pods running on node1 |
+| --node=""            | kubectl get pods --field-selector spec.nodeName= | Fetch all pods without an assigned node |
```
Comment (dgrisonnet) [inline README.md:302]: Similar to my comment in https://github.com/kubernetes/kube-state-metrics/pull/2373#issuecomment-2066910515, I don't think that specifying `--node=""` is very user friendly.

We could either implicitly do it, or put it in a dedicated flag such as `--fetch-unscheduled-pods`

Comment (CatherineF-dev) [reply README.md]: Fixed the documentation to `--node=` instead of `--node=""`. Will it be clearer? Or we prefer `--fetch-unscheduled-pods`

**Comment by mrueg** [comment]
Let's add this once https://github.com/kubernetes/kube-state-metrics/pull/2388 is in.

We can refactor the behavior later.

**Comment by CatherineF-dev** [comment]
This PR is not needed due to https://github.com/kubernetes/kube-state-metrics/pull/2388

**Pull request closed without merging**